pandas/core/groupby/generic.py

```diff
@@ -1357,7 +1357,13 @@ def arr_func(bvalues: ArrayLike) -> ArrayLike:
 
         # We could use `mgr.apply` here and not have to set_axis, but
         #  we would have to do shape gymnastics for ArrayManager compat
-        res_mgr = mgr.grouped_reduce(arr_func, ignore_failures=True)
+        try:
+            res_mgr = mgr.grouped_reduce(
+                arr_func, ignore_failures=numeric_only is lib.no_default
+            )
+        except NotImplementedError as err:
+            # For NotImplementedError, args[0] is the error message
```

Comment:
is this still the case if it was raised with just `raise NotImplementedError`?

Reply:
Ah - no, it's not. I'll fix in a followup. Thanks.